File: 1290-ConvertBinaryNumberinaLinkedListtoInteger/1290-ConvertBinaryNumberinaLinkedListtoInteger.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {number}
 */
var getDecimalValue = function(head) {
    let curr = head;
    let ans = curr.val;

    while(curr && curr.next){
        ans = (ans * 2) + curr.next.val;
        curr = curr.next;
    }
    return ans;
};

File: 1290-ConvertBinaryNumberinaLinkedListtoInteger/1290-ConvertBinaryNumberinaLinkedListtoInteger.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def getDecimalValue(self, head: ListNode) -> int:
        curr = head        
        ans = curr.val

        while curr and curr.next:
            ans = (ans * 2) + curr.next.val
            
            curr = curr.next
        
        return ans
            
